apps/mobile/src/libs/components/wheel/libs/components/animated-pulsing-sector/animated-pulsing-sector.tsx

```diff
@@ -0,0 +1,177 @@
+import { withTiming } from "react-native-reanimated";
+
+import { Animated, Circle } from "~/libs/components/components";
+import {
+	useAnimatedProps,
+	useEffect,
+	useSharedValue,
+} from "~/libs/hooks/hooks";
+
+import { getRandomValue, getSectorParameters } from "../../helpers/helpers";
+
+type Properties = {
+	animationRepetitions: number;
+	animationTime: number;
+	centerGap: number;
+	centerPoint: number;
+	endPercentInner: number;
+	endPercentOuter: number;
+	height: number;
+	layerOffset: number;
+	maxHeight: number;
+	outlineColor: string;
+	sectorColor: string;
+	startPercentInner: number;
+	startPercentOuter: number;
+};
+
+const HALF_VALUE_COEFFICIENT = 0.5;
+
+const AnimatedCircle = Animated.createAnimatedComponent(Circle);
+
+const AnimatedPulsingSector: React.FC<Properties> = ({
+	animationRepetitions,
+	animationTime,
+	centerGap,
+	centerPoint,
+	endPercentInner,
+	endPercentOuter,
+	height,
+	layerOffset,
+	maxHeight,
+	outlineColor,
+	sectorColor,
+	startPercentInner,
+	startPercentOuter,
+}: Properties) => {
+	const {
+		innerDashArrayDash,
+		innerDashArrayGap,
+		innerDashOffset,
+		innerStrokeWidth,
+		outerDashArrayDash,
+		outerDashArrayGap,
+		outerDashOffset,
+		outerStrokeWidth,
+		radius,
+	} = getSectorParameters({
+		centerGap,
+		endPercentInner,
+		endPercentOuter,
+		height,
+		layerOffset,
+		startPercentInner,
+		startPercentOuter,
+	});
+
+	const animatedInnerArrayDash = useSharedValue(innerDashArrayDash);
+	const animatedInnerArrayGap = useSharedValue(innerDashArrayGap);
+	const animatedInnerDashOffset = useSharedValue(innerDashOffset);
+	const animatedInnerStrokeWidth = useSharedValue(innerStrokeWidth);
+
+	const animatedOuterArrayDash = useSharedValue(outerDashArrayDash);
+	const animatedOuterArrayGap = useSharedValue(outerDashArrayGap);
+	const animatedOuterDashOffset = useSharedValue(outerDashOffset);
+	const animatedOuterStrokeWidth = useSharedValue(outerStrokeWidth);
+
+	const animatedRadius = useSharedValue(radius);
+	const timingSettings = { duration: animationTime };
+
+	useEffect(() => {
+		const nextHeight = getRandomValue({
+			max: maxHeight,
+			min: maxHeight * HALF_VALUE_COEFFICIENT,
+		});
+		const {
```

Comment:
Can we move all this logic to a separate hook?

Reply:
I agree, some custom hook like `useWheelAnimation` can work here